jujuclient/file.go

```diff
@@ -629,6 +629,13 @@ func (s *store) UpdateCredential(cloudName string, details cloud.CloudCredential
 	}
 
 	all[cloudName] = details
+
+	// If cloud has no credentials now, delete cloud references from
+	// credentials file as well.
+	if len(all[cloudName].AuthCredentials) == 0 {
```

Comment:
rather than assigning to all (above on line 631), then checking it and maybe deleting from it:

```go
if len(details.AuthCredentials) > 0 {
    all[cloudName] = details
} else {
    delete(all, cloudName)
}
```